src/services/codefixes/inferFromUsage.ts

```diff
@@ -414,27 +414,33 @@ namespace ts.codefix {
                 inferTypeFromContext(reference, checker, usageContext);
             }
             const isConstructor = declaration.kind === SyntaxKind.Constructor;
-            const callContexts = isConstructor ? usageContext.constructContexts : usageContext.callContexts;
-            return callContexts && declaration.parameters.map((parameter, parameterIndex): ParameterInference => {
-                const types: Type[] = [];
+            const isJS = isInJSFile(declaration);
+            const callContexts = isJS ? [...usageContext.constructContexts || [], ...usageContext.callContexts || []] :

```

Comment:
You could just combine construct and call unconditionally?

Reply:
Doesn't make a difference when run on the user test suite, and it simplifies the code considerably, so I think it's a good change.